examples/apps/echo/client/echo_client.py

```diff
@@ -166,36 +172,61 @@ def Main(args=None):
 	requester_id = secrets.token_hex(32)
 	session_iv = utility.generate_iv()
 	session_key = utility.generate_key()
-	encrypted_session_key = utility.generate_encrypted_key(
-		session_key, worker_obj.encryption_key)
-	requester_nonce = secrets.token_hex(32)
+	requester_nonce = secrets.token_hex(16)
 	# Create work order
-	wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "JSON-RPC",
+	wo_submit_json = WorkOrderParams(
 		work_order_id, worker_id, workload_id, requester_id,
-		encrypted_session_key, session_iv, requester_nonce,
-		worker_encryption_key=base64.b64decode(
-			worker_obj.encryption_key).hex(),
+		session_key, session_iv, requester_nonce,
+		result_uri=" ", notify_uri=" ",
+		worker_encryption_key=worker_obj.encryption_key,
 		data_encryption_algorithm="AES-GCM-256")
+	# Add worker input data
 	wo_submit_json.add_in_data(message)
 
 	# Sign work order
 	private_key = utility.generate_signing_keys()
-
+	wo_submit_json.add_encrypted_request_hash()
+	wo_submit_json.add_requester_signature(private_key)
 	# Submit work order
-	direct_wrapper.init_work_order(config)
-	direct_wrapper.work_order_submit(wo_submit_json, encrypted_session_key, 
-		worker_obj, private_key, session_key, session_iv)
+	logger.info("Work order submit request : %s, \n \n ",
+        wo_submit_json.to_string())
+	req_id += 1
+	direct_jrpc.create_work_order_adaptor(
+		config).work_order_submit(
+			wo_submit_json.get_params(),
+			wo_submit_json.get_in_data(),
+			wo_submit_json.get_out_data(),
+			id=req_id
+		)
 
 	# Retrieve result
-	wo_get_result_json = jrpc_request.WorkOrderGetResultJson(4, work_order_id)
-	direct_wrapper.work_order_get_result(
-		wo_get_result_json, session_key, session_iv)
-
+	req_id += 1
+	res = direct_jrpc.create_work_order_adaptor(
+		config).work_order_get_result(
+			work_order_id,
+			req_id
+		)
+	while("result" not in res):
```

Comment:
Add a function is_result_ready() in direct_jrpc class which client can use.

Reply:
"result" is part of response dictionary, initially response will have "error" in response dictionary initially and later once result available then it will response dictionary.

Reply:
framework should hide these complexities from client. One way to do this is to implement get_work_order_status and get_work_order_result. If result is ready then client can call get_work_order_result